spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -432,4 +405,340 @@ public void testAggregateExpressions() {
     assertThat(r1.get(10)).isEqualTo(0.25); // VAR_POP(num1)
     assertThat(r1.get(11)).isEqualTo(0.5); // VAR_SAMP(num1)
   }
+
+  @Test
+  public void testJoinQuery() {
+    writeTestDataToBigQuery();
+    Dataset<Row> df =
+        spark
+            .read()
+            .format("bigquery")
+            .option("materializationDataset", testDataset.toString())
+            .load(testDataset.toString() + "." + testTable);
+
+    df.createOrReplaceTempView("numStructDF");
+
+    // Creating a data frame of schema NumStruct, and writing it to BigQuery to perform the join
+    // using pushdown
+    Dataset<Row> df_to_join =
+        spark
+            .createDataset(
+                Arrays.asList(
+                    new NumStruct(
```

Comment:
Lets move all of this out to a static variable in the class

Reply:
Created a method similar to the implementation in WriteIntegrationTests